### PR TITLE
update match content

### PR DESCRIPTION
**Pull request by Ibrahim-Jarada**
Relates #48

### CODE REVIEW

File: src/controllers/users/matchUserContent.ts [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
   try {
+    if (body.recoveredCosts > (body.filmingCosts + body.advance + body.feePaid)) throw new CustomError("Recovered costs can't be greater than costs", 400);
+
+    const modifiedBody = {
+      ...body,
+      filmingCosts: body.filmingCosts.toString(),
+      advance: body.advance.toString(),
+      feePaid: body.feePaid.toString(),
+      recoveredCosts: body.recoveredCosts.toString(),
+    };
```
Comment (Jood80) [inline src/controllers/users/matchUserContent.ts:17]: When do you start thinking that this object shouldn't be here?
despite the fact that you should only include your pure logic in your controller -without any service explicitly implemented "just to be called"-.  any objects that has to be processed before being used in the code, either to convert its type or to make any sort of modification over it should be moved onto `dto` object.
your controller should be as clean as possible, so others can quickly grasping what's the logic behind it

File: src/controllers/users/matchUserContent.ts [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
   const { body } = req;
-  const modifiedBody = {
-    ...body,
-    filmingCosts: body.filmingCosts.toString(),
-    advance: body.advance.toString(),
-    feePaid: body.feePaid.toString(),
-  };
 
   try {
+    if (body.recoveredCosts > (body.filmingCosts + body.advance + body.feePaid)) throw new CustomError("Recovered costs can't be greater than costs", 400);
```
Comment (Jood80) [inline src/controllers/users/matchUserContent.ts:9]:
```suggestion
const remainingCost = body.filmingCosts + body.advance + body.feePaid
    if (body.recoveredCosts > remainingCost ) throw new CustomError("Recovered costs can't be greater than costs", 400);
```
if those accumalted values aren't `remaingCost` and instead are `accumulatedCost` or `totalCost` then pick whatever you see is more accurate and rename it

**Review by Jood80 — CHANGES_REQUESTED** [review]

File: src/services/user/matchUserContent.ts [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
+import { EventEmitter } from 'node:events';
 import { CustomError } from '../../helpers';
 import { HttpStatus } from '../../helpers/constants';
 
+let settings = getDashboardSettings();
+class MyEmitter extends EventEmitter {}
+
+const myEmitter = new MyEmitter();
+myEmitter.on('update', () => {
+  console.log('settings was updated');
```
Comment (Jood80) [inline src/services/user/matchUserContent.ts:18]: `logger.success` maybe?

File: src/services/user/matchUserContent.ts [hunk 4]
```diff
@@ -18,18 +35,116 @@ const matchUserContent: IMatchUserContent = async ({
...
+    watchedSeconds,
+    revenue,
+    createdBy,
+    updatedBy,
+    tvodTicketsCount,
+    tvodSeconds,
+    watches,
+    report,
+  }) => {
+    if (!report) throw Error();
```
Comment (Jood80) [inline src/services/user/matchUserContent.ts:76]: that `Error()` has to be updated with what we're using as a `CustomError` for now

File: src/services/user/matchUserContent.ts [hunk 5]
```diff
@@ -1,14 +1,31 @@
...
 
+let settings = getDashboardSettings();
+class MyEmitter extends EventEmitter {}
+
+const myEmitter = new MyEmitter();
+myEmitter.on('update', () => {
+  console.log('settings was updated');
+  settings = getDashboardSettings();
+});
 type IMatchUserContent = (_: {
```
Comment (Jood80) [inline src/services/user/matchUserContent.ts:21]: please update this as @Mohammed-Hamada suggested in [his issue](https://github.com/NextUp-comedy/api-nc/issues/44)

**Review by Jood80 — APPROVED** [review]
Nicely done Ibrahim! 
it's impressive to see you carefully covering these cases!
I would suggest to add some sort of comments in your `matchUserContent` service tho, it will make it more comprehensible.

**Review by zaher-aa — APPROVED** [review]

**Review by RandSohail — APPROVED** [review]

**Review by AymanSami98 — APPROVED** [review]

**Pull request merged**